src/core/api/repository.go

```diff
@@ -463,6 +463,16 @@ func (ra *RepositoryAPI) Retag() {
 		return
 	}
 
+	if !utils.ValidateRepo(repo) {
```

Comment:
move the validation up to line 437, it should be the first action in the API.

Reply:
done